genome_designer/genome_finish/tests/test_parallel_assembly.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.test import TestCase
+
+from genome_finish.assembly import run_de_novo_assembly_pipeline
+from genome_finish.millstone_de_novo_fns import create_de_novo_variants_set
+from main.model_utils import get_dataset_with_type
+from main.models import Dataset
+from main.models import ExperimentSample
+from main.models import ExperimentSampleToAlignment
+from main.models import Project
+from main.testing_util import are_fastas_same
+from pipeline.pipeline_runner import run_pipeline
+from utils.import_util import add_dataset_to_entity
+from utils.import_util import import_reference_genome_from_local_file
+from utils.reference_genome_maker_util import generate_new_reference_genome
+
+TEST_USERNAME = 'testuser'
+TEST_PASSWORD = 'password'
+TEST_EMAIL = 'test@example.com'
+
+GF_TEST_DIR = os.path.join(
+        settings.PWD,
+        'test_data/genome_finish_test')
+
+
+class TestParallelAssembly(TestCase):
```

Comment:
FYI - Django tests use celery in synchronous mode. So this test is good for making sure semantics work, but it won't catch concurrency issues.

I tried setting up integration tests in the pasts (https://github.com/churchlab/millstone/blob/master/genome_designer/tests/integration/test_pipeline_integration.py#L31). These would bring up a celery server and run things in parallel. However, it caused way too many headaches and cryptic problems so we abandoned that route.